File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
         return false;  
        int revNum = 0;
        int dup = x;
        
        while(x > 0) {
            int ld = x % 10;
            revNum = (revNum * 10) + ld;
            x = x / 10;
        }
        
        return dup == revNum;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int n) {
        if (n < 0) return false;

        int dup = n;
        long long revdigit = 0;

        while (n > 0) {
            int last_digit = n % 10;
            revdigit = revdigit * 10 + last_digit;
            n = n / 10;
        }

        return revdigit == dup;
    }
};
